File: bot/python/commands/ban.py
""" 
 * vACS Project
 * Copyright (C) 2020-2023 Sympact06 & Stuncs69 & TheJordinator
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 * For more information about the vACS Project, visit:
 * https://github.com/sympact06/vacs
 
 """


from concurrent.futures import ThreadPoolExecutor
from os import name
from nextcord.ext import commands
import nextcord, asyncio, requests


class BanHandler(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @nextcord.slash_command(description="Ban Handler")
    async def ban(self, interaction: nextcord.Interaction):
        pass

    @ban.subcommand(description="View someones ban")
    async def view(
        self,
        interaction: nextcord.Interaction,
        roblox: int = nextcord.SlashOption(name="roblox_id"),
    ):
        req = requests.get(
            "https://api.domain.com/v1/roblox/vacs/blacklist/get",
            json={"token": "SECRET", "rbxid": f"{roblox}"},
        )

        data = req.json()

        if data["success"]:
            data = data["ban_data"]
            embed = nextcord.Embed(title=data["rbxid"] + " | " + data["ban_type"])
            embed.add_field(name="Reason", value=data["reason"])
            embed.set_footer(text="Banned at " + data["banned_at"])

            await interaction.send(embed=embed, ephemeral=True)
        else:
            await interaction.send(content="User is not banned on vACS", ephemeral=True)

    @ban.subcommand(description="Add ban to user")
    async def add(
        self,
        interaction: nextcord.Interaction,
        type: str = nextcord.SlashOption(
            name="type",
            description="The type of ban given",
            choices=["A", "B", "C"],
            required=True,
        ),
        rs: str = nextcord.SlashOption(name="reason", required=True),
        id: int = nextcord.SlashOption(name="roblox_id", description="The ROBLOX id"),
    ):
        if interaction.user.guild_permissions.ban_members:
            req = requests.post(
                "https://api.domain.com/v1/roblox/vacs/blacklist/add",
                json={
                    "token": "SECRET",
                    "ban_type": type,
                    "reason": rs,
                    "rbxid": f"{id}",
                },
            )
            await interaction.send(content=req.json()["message"], ephemeral=True)
        else:
            await interaction.send(
                content="You do not have permission to use this command", ephemeral=True
            )

    @ban.subcommand(description="Remove ban from user")
    async def remove(
        self,
        interaction: nextcord.Interaction,
        type: int = nextcord.SlashOption(name="roblox_id", required=True),
    ):
        if interaction.user.guild_permissions.ban_members:
            req = requests.post(
                "https://api.domain.com/v1/roblox/vacs/blacklist/remove",
                json={"token": "SECRET", "rbxid": f"{type}"},
            )
            await interaction.send(content=req.json()["message"], ephemeral=True)
        else:
            await interaction.send(
                content="You do not have permission to use this command", ephemeral=True
            )


def setup(bot):
    bot.add_cog(BanHandler(bot))

File: www/lua/ban.php
<?php
/*
 * vACS Project
 * Copyright (C) 2020-2023 Sympact06 & Stuncs69 & TheJordinator
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published
 * by the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 *
 * For more information about the vACS Project, visit:
 * https://github.com/sympact06/vacs
*/

date_default_timezone_set('Europe/Amsterdam');
class BansAPI
{
    private $db;

    public function __construct()
    {
        $db_host = 'localhost';
        $db_name = 'vACS';
        $db_user = 'root';
        $db_pass = '';
        $this->db = new PDO("mysql:host=$db_host;dbname=$db_name", $db_user, $db_pass);
    }

    public function add_ban($token, $rbxid)
    {
        if (!$this->is_token_valid($token))
        {
            return array(
                'success' => false,
                'message' => 'Invalid token'
            );
        }

        $stmt = $this
            ->db
            ->prepare("INSERT INTO bans (ban_type, banned_at, reason, rbxid) VALUES (:ban_type, :banned_at, :reason, :rbxid)");
        $banned_at = date('Y-m-d H:i:s');
        $ban_type = "A";
        $reason = "Automated Anti-Cheat Ban";

        $stmt->bindParam(':ban_type', $ban_type);
        $stmt->bindParam(':banned_at', $banned_at);
        $stmt->bindParam(':reason', $reason);
        $stmt->bindParam(':rbxid', $rbxid);

        if ($stmt->execute())
        {
            return array(
                'success' => true,
                'message' => 'Ban added successfully'
            );
        }
        else
        {
            return array(
                'success' => false,
                'message' => 'Failed to add ban'
            );
        }
    }

    private function is_token_valid($token)
    {

        if ($token == 'SECRET')
        {
            return true;
        }
        else
        {
            return false;
        }

    }
}

$rbxid = $_GET['rbxid'];
$token = $_GET['token'];

$bans_api = new BansAPI();

$response = $bans_api->add_ban($token, $rbxid);

echo json_encode($response);
?>
